specs/data_structures.md

```diff
@@ -24,50 +23,63 @@ Data Structures
 - [Merkle Tree](#merkle-tree)
   - [Namespace Merkle Tree](#namespace-merkle-tree)
 - [Erasure Coding](#erasure-coding)
+  - [AvailableHeader](#availableheader)
+  - [TransactionData](#transactiondata)
+  - [MessageData](#messagedata)
+  - [Transaction](#transaction)
+  - [Message](#message)
 
 # Blockchain Data Structures
 
 ## Block
 
-| name         | type                          | description |
-| ------------ | ----------------------------- | ----------- |
-| `header`     | [Header](#header)             |             |
-| `data`       | [Data](#data)                 |             |
-| `evidence`   | [EvidenceData](#evidencedata) |             |
-| `lastCommit` | [Commit](#commit)             |             |
+Blocks are the top-level data structure of the LazyLedger blockchain.
+
+| name            | type                            | description                                                           |
+| --------------- | ------------------------------- | --------------------------------------------------------------------- |
+| `header`        | [Header](#header)               | Block header. Contains primarily identification info and commitments. |
+| `availableData` | [AvailableData](#availabledata) | Data that is erasure-coded for availability.                          |
+| `evidence`      | [EvidenceData](#evidencedata)   | Evidence used for slashing conditions (e.g. equivocation).            |
+| `lastCommit`    | [Commit](#commit)               | Previous block's Tendermint commit.                                   |
 
 ## Header
 
-| name                 | type                      | description |
-| -------------------- | ------------------------- | ----------- |
-| `version`            | [Version](#version)       |             |
-| `chainID`            | `uint64`                  |             |
-| `height`             | `uint64`                  |             |
-| `time`               | [Time](#time)             |             |
-| `lastBlockID`        | [BlockID](#blockid)       |             |
-| `lastCommitHash`     | [HashDigest](#hashdigest) |             |
-| `dataHash`           | [HashDigest](#hashdigest) |             |
-| `validatorsHash`     | [HashDigest](#hashdigest) |             |
-| `nextValidatorsHash` | [HashDigest](#hashdigest) |             |
-| `consensusHash`      | [HashDigest](#hashdigest) |             |
-| `appHash`            | [HashDigest](#hashdigest) |             |
-| `lastResultsHash`    | [HashDigest](#hashdigest) |             |
-| `evidenceHash`       | [HashDigest](#hashdigest) |             |
-| `proposerAddress`    | [Address](#address)       |             |
-
-## Data
-
- | name  | type                            | description |
- | ----- | ------------------------------- | ----------- |
- | `txs` | [Transaction](#transaction)`[]` |             |
-
-TODO define a transaction format
+Block header, which is fully downloaded by both full clients and light clients.
```

Comment:
Note we're considering some changes that might simplify the header: https://github.com/tendermint/tendermint/issues/2530. Would be cool to have your input.